frontend/ui/widget/imageviewer.lua

```diff
@@ -811,9 +810,8 @@ function ImageViewer:onSaveImageView()
         self:update()
         UIManager:forceRePaint()
     end
-    local screenshots_dir = G_reader_settings:readSetting("screenshot_dir") or DataStorage:getDataDir() .. "/screenshots/"
-    local screenshot_name = os.date(screenshots_dir .. "ImageViewer" .. "_%Y-%m-%d_%H%M%S.png")
-    UIManager:sendEvent(Event:new("Screenshot", screenshot_name, restore_settings_func))
+    local screenshot_name = os.date(Screenshoter:getScreenshotDir() .. "/ImageViewer_%Y-%m-%d_%H%M%S.png")
+    Screenshoter:onScreenshot(screenshot_name, restore_settings_func)
```

Comment:
Feels a bit odd to call this method on the module/class object and not an instance. Might not be an issue, there's not much use of `self` in it, ~~except for this assignment: `self.screenshot_fn_fmt = screenshots_dir .. self.prefix .. "_%Y-%m-%d_%H%M%S.png"` which will pollute the class, but will not be used anywhere (so it may as well be a `local`).~~
edit: I see you removed `self.screenshot_fn_fmt` 

Dunno if when using `UIManager:sendEvent()` it's a real instance that get it called. If yes, your optimisation to avoid UIManager:sendEvent() may not be worth the oddness here.
edit: what about `file = self.ui and self.ui.document and self.ui.document.file` in Screenshoter:onScreenshot when `self` is not an instance?

Reply:
Yes, the instance must be used, to allow setting an image from the book opened in ImageViewer as book cover.
Reverted to event.